Sources/Commands/Workspace.swift

```diff
@@ -301,6 +304,10 @@ public class Workspace {
         try localFileSystem.createDirectory(repositoriesPath, recursive: true)
         try localFileSystem.createDirectory(checkoutsPath, recursive: true)
 
+        // Create a file lock and immediately try to aquire the lock before restoring the previous state.
+        fileLock = FileLock(name: "swiftpm-workspace-lock", cachePath: self.dataPath)
+        try fileLock.lock()
```

Comment:
Can we get a timeout on this, and then diagnostics? I hate it when tools just hang and don't ever tell you why.